src/polyfem/State.cpp

```diff
@@ -1030,9 +1091,9 @@ namespace polyfem
 
 		out_geom.build_grid(*mesh, args["output"]["advanced"]["sol_on_grid"]);
 
-		if (!problem->is_time_dependent() && boundary_nodes.empty())
+		if ((!problem->is_time_dependent() || args["time"]["quasistatic"]) && boundary_nodes.empty())
 		{
-			log_and_throw_error("Static problem need to have some Dirichlet nodes!");
+			logger().warn("(Quasi-)Static problem without Dirichlet nodes, will fix solution at one node to find a unique solution!");
```

Comment:
why we have quasi static, what is the difference?

Reply:
Quasi-static has been there for a long time, it's transient without inertia, but I guess no one was using it. I added quasi-static here because the linear system is also singular just as static.